ctx.go

```diff
@@ -235,6 +235,43 @@ func (c *Ctx) UsePrivateKey(key PrivateKey) error {
 	return nil
 }
 
+// UserCertAndNullKey configures the context to use the given certificate
+// and NULL private key for the SSL handshakes.
+// It allows you to use private keys that are never accessible directly
+// e.g.: to which openssl has access only via Engine module.
+// https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_use_cert_and_key.html
+func (c *Ctx) UseCertAndNullKey(cert *Certificate) error {
+	if cert == nil {
+		return errors.New("cert cant be nil")
```

Comment:
```suggestion
		return errors.New("no certificate given")
```
mby? or received? Hmm :thinking:

Reply:
fixed.